chalk-integration/src/program.rs

```diff
@@ -126,6 +157,14 @@ impl tls::DebugContext for Program {
         )
     }
 
+    fn debug_opaque_ty(
+        &self,
+        opaque_ty: &OpaqueTy<ChalkIr>,
+        fmt: &mut fmt::Formatter<'_>,
+    ) -> Result<(), fmt::Error> {
+        write!(fmt, "impl {:?}", opaque_ty.opaque_ty_id)
```

Comment:
...and then reference the `opaque_type_names` here to print the *name* of the opaque type, and not "impl"